fields.fpp

```diff
@@ -980,31 +1364,13 @@ contains
       if (dist == 'h') then
          phi = phi / gamtot_h
       else if (dist == 'gbar') then
-         if (full_flux_surface) then
-!        ! need to invert 1-gamma0 operator
-!        allocate (phi_swap(naky_all,ikx_max))
-!        do it = 1, ntubes
-!           do iz = -nzgrid, nzgrid
-!              call swap_kxky_ordered (phi, phi_swap)
-!              do ikx = 1, ikx_max
-!                 ! if ky values uncoupled, then obtain phi by simple divide
-!                 if (maxval(ia_max_gam0a(:,ikx,iz)) == 1) then
-!                    phi_swap(:,ikx) = phi_swap(:,ikx)/gam0a(1,:,ikx,iz)
-!                 else
-!                    ! for ky values that require info from other ky values
-!                    ! must solve linear system
-!                    call
-!                 end if
-!              end do
-!           end do
-!        end do
-!        deallocate (phi_swap)
-         else if ((radial_variation .and. ky_solve_radial > 0 &
-                   .and. runtype_option_switch /= runtype_multibox) &
-                  .or. &!DSO -> sorry for this if statement
-                  (radial_variation .and. ky_solve_radial > 0 .and. job == 1 &
-                   .and. runtype_option_switch == runtype_multibox &
-                   .and. .not. ky_solve_real)) then
+         if ((radial_variation .and. ky_solve_radial > 0 &
+              .and. runtype_option_switch /= runtype_multibox) &
+             .or. &!DSO -> sorry for this if statement
+             (radial_variation .and. ky_solve_radial > 0 .and. job == 1 &
+              .and. runtype_option_switch == runtype_multibox &
+              .and. .not. ky_solve_real)) then
```

Comment:
One way to deal with complex conditionals like this is to use some local intermediate variables to store parts/all of the expression. These can use "nicer" names, which combine to make the whole thing more readable. Here's a (very naïve) suggestion:

```suggestion
         has_radial_variation = (radial_variation .and. ky_solve_radial > 0)
         is_multibox_run = (runtype_option_switch == runtype_multibox)
         is_first_multibox_job = is_multibox_run .and. (job == 1)

         if ((has_radial_variation .and. .not. is_multibox_run) &
             .or. (has_radial_variation .and. is_first_multibox_job &
                   .and. .not. ky_solve_real)) then
```

This takes up a bit more space, but can make the whole thing easier to read (and therefore easier to check!). Picking good names can be tricky, of course.

Reply:
these conditionals are not part of the changes made in this branch, so I'll leave them for DSO to comment

Reply:
Yeah, I can fix this, along with cleaning up fields.fpp where I clobbered it with radial variation changes. I think this should be a separate pull request, however, since the changes may not be insignificant.

Reply:
Ah sorry, I see this was just moved rather than added. Yes, a separate PR sounds good.